### ISSUE TITLE
Discord mime handler registration broken in snap packages

### ISSUE THREAD

**Issue by elementc** [bitfighter/bitfighter#640]
Am aware this is a bug I introduced by adding the package type. Writing it down so this doesn't get lost.

**BF version:** master since https://github.com/bitfighter/bitfighter/pull/638
**OS/arch:** Linux (snap package)

**Describe the bug**
On starting up the snap build of the game, the following error message appears in stderr:
```
casey on IRONFIRE in ~
23:19:31 🍀  bitfighter
sh: 1: xdg-mime: not found
Welcome to Bitfighter!
```
The xdg-mime command comes from the discord_register_linux.cpp file:
https://github.com/bitfighter/bitfighter/blob/5d071faaf86e6eb32dc4f2bfbe86d0711703f984/discord-rpc/src/discord_register_linux.cpp#L85-L93

By design snapped applications can't escape their confinement, so simply adding xdg-mime to the runtime packages for our snap will not do anything meaningful- just register a dead-end mime type inside of an isolated container. While [there's been discussion of a way to add mime registration support to snap](https://forum.snapcraft.io/t/allow-snaps-to-register-new-mime-types/6467),  and [apps can already register as handler for an existing mime type via the `MimeType=` key in their .desktop file](https://developer.gnome.org/integration-guide/stable/mime.html.en), snapped apps can't yet register new mime types.

I don't know much about the discord rpc sdk, but under the hood it's [supposed to be just opening a local websocket on a well-known port](https://discord.com/developers/docs/topics/rpc). Indeed it seems to have done so on my system:

```
casey on IRONFIRE in ~ 
at 23:57:41 🔥  netstat -tulpn | grep :64.. 
tcp        0      0 127.0.0.1:6463          0.0.0.0:*               LISTEN      288508/mainScreenPr 

casey on IRONFIRE in ~ 
at 23:57:59 🍀  ps aux | grep 288508
casey     288508  7.4  0.8 12756896 285696 ?     Sl   Oct18 435:09 /usr/share/discord/Discord <snipped stack of args>
```

Since there's no special unix-domain sockets or other exotic IPC under the hood, I think discord should still work fine with the network permissions we already have on the snap, so since the mime handler is what lets discord launch bitfighter directly, we may need to find some other method to get that MIME handler registered (and fix the code that's fruitlessly trying to register it). Maybe the main .desktop file can be modified to at least list the contents that the code is trying to add in a generated .desktop file here:
https://github.com/bitfighter/bitfighter/blob/5d071faaf86e6eb32dc4f2bfbe86d0711703f984/discord-rpc/src/discord_register_linux.cpp#L44-L50

`bitfighter` wouldn't be able to set itself as the default handler for this mime type, but at least it could be *a* handler for it.

Additional note while looking at this code snippet:
```cpp
    if (system(xdgMimeCommand) < 0) {
        fprintf(stderr, "Failed to register mime handler\n");
    }
```
I think this condition is incorrect. We fail to register the mime handler on any nonzero return value, not just negative values.

**Comment by raptor** [comment]
I have no idea what to do here. We could modify the discord-rpc code, but it's third-party and I'd like to avoid that. (It's also been deprecated in favor of a new API that doesn't give the source code [Discord GameSDK]).

I don't really want to modify bitfighter to handle discord weirdness stuff too much - there's been a long history of 3rd-party integrations that disappear over time and I'd like to keep it as independent as possible.

**Comment by elementc** [comment]
@raptor discord works fine in AppImage, I think snap is a dead end. Want me to revert?

**Comment by raptor** [comment]
Nah...  the .yaml file can stay as a reference for others.  Mind if I close this?

**Comment by elementc** [comment]
Sounds good to me.

**Comment by elementc** [comment]
heads up to @Mailaender (and sorry for all the pings, this is my first time working with the flatpak build in depth) that this bug also happens in the Flatpak build:
```
casey on WATERFALCON in org.bitfighter.Bitfighter on  master took 3s 
at 15:19:20 🍀 flatpak run org.bitfighter.Bitfighter 
Copying resources
sh: line 1: xdg-mime: command not found
Welcome to Bitfighter!
```

**Comment by Mailaender** [comment]
I don't know why `xdg-mime` is not part of the FreeDesktop runtime.

**Comment by elementc** [comment]
> `xdg-mime` is probably not part of the FreeDesktop runtime, because it won't work in the sandbox environment.

Yes, that was the conclusion we came to in the snap world. I am going to reopen this bug: I think we can make this not error out in that way, and I think there is a way in flatpak to register the appropriate MIME handler for Discord to be able to launch us, based on https://github.com/flatpak/flatpak/issues/1572

**Comment by Mailaender** [comment]
I am not sure if https://github.com/flathub/com.github.Anuken.Mindustry/commit/3307847e9697a862f06ef776fc0b459e8a699daf would help in this case.
